openpype/modules/ftrack/plugins/publish/collect_ftrack_family.py

```diff
@@ -1,29 +1,108 @@
+"""
+Requires:
+    none
+
+Provides:
+    instance     -> families ([])
+"""
 import pyblish.api
+import avalon.api
+
+from openpype.lib.plugin_tools import filter_profiles
+
+
+class CollectFtrackFamily(pyblish.api.InstancePlugin):
+    """
+        Adds explicitly 'ftrack' to families to upload instance to FTrack.
 
+        Uses selection by combination of hosts/families/tasks names via
+        profiles resolution.
 
-class CollectFtrackFamilies(pyblish.api.InstancePlugin):
-    """Collect family for ftrack publishing
-  
-    Add ftrack family to those instance that should be published to ftrack
+        Triggered everywhere, checks instance against configured.
 
+        Checks advanced filtering which works on 'families' not on main
+        'family', as some variants dynamically resolves addition of ftrack
+        based on 'families' (editorial drives it by presence of 'review')
     """
+    label = "Collect Ftrack Family"
+    order = pyblish.api.CollectorOrder + 0.4998
 
-    order = pyblish.api.CollectorOrder + 0.3
-    label = 'Add ftrack family'
-    families = ["model",
-                "setdress",
-                "model",
-                "animation",
-                "look",
-                "rig",
-                "camera"
-                ]
-    hosts = ["maya"]
+    profiles = None
 
     def process(self, instance):
+        if not self.profiles:
+            self.log.warning("No profiles present for adding Ftrack family")
+            return
+
+        anatomy_data = instance.context.data["anatomyData"]
+        task_name = instance.data.get("task",
+                                      avalon.api.Session["AVALON_TASK"])
+        host_name = anatomy_data.get("app",
+                                     avalon.api.Session["AVALON_APP"])
+        family = instance.data["family"]
+
+        filtering_criteria = {
+            "hosts": host_name,
+            "families": family,
+            "tasks": task_name
+        }
+        profile = filter_profiles(self.profiles, filtering_criteria)
+
+        if profile:
```

Comment:
I would convert this condition to oposite logic. If not profile then log and return to prevent uncessary indentation in main logic.

Reply:
I am not big fan of unnecessary multiple return statements, I can live with ripcord return at the beginning of 'process', I can see use of them with deeper structure, but with '1 level if' I don't see much use nor beauty there, imho.

Reply:
I am because if somebody will add some lines afterwards in future he can easily miss that if there are no `profiles` he should do nothing. With `return` you are explicit saying that there is nothing to do...